### PR TITLE
[Term Entry] General: REST API

### PR DESCRIPTION
**Pull request by caupolicandiaz**
### Description

Adding a general term entry for the REST architecture.

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [x ] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [ x] I have performed a self-review of my own writing and code.
- [ x] I have checked my entry and corrected any misspellings.
- [ x] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by HishamT — APPROVED** [review]
LGTM!

File: content/general/concepts/rest/rest.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - 'Dependency'
+  - 'Interface'
+CatalogContent:
+  - 'paths/computer-science'
+  - 'paths/front-end-engineer-career-path'
+---
+
+**REST** is an architectural style for [API](https://www.codecademy.com/resources/docs/general/api) design that is defined by six constraints. These constraints were first documented in Roy Fielding's dissertation ["Architectural Styles and the Design of Network-based Architectures"](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm). The acronym REST stands for REpresentational State Transfer, which is foundational to this design architecture, the sharing of resources, or rather representations of those resources, across systems.
+
+The constraints that define REST architecture include the following:
```
Comment (yangc95) [inline content/general/concepts/rest/rest.md:19]:
```suggestion
## REST Constraints
```

Could we change this to a header, feel free to adjust~

**Review by yangc95 — COMMENTED** [review]
@caupolicandiaz Just have one suggestion~

File: content/general/concepts/rest/rest.md (hunk 1 above)
Comment (caupolicandiaz) [reply content/general/concepts/rest/rest.md]: > Could we change this to a header, feel free to adjust~

All set!

**Review by yangc95 — APPROVED** [review]

**Review by SSwiniarski — APPROVED** [review]
@caupolicandiaz Looks good.

**Comment by yangc95** [comment]
@caupolicandiaz live link: https://www.codecademy.com/resources/docs/general/rest

Screenshot 2023-03-29 at 12 28 13 PM

cc: @SSwiniarski

**Pull request merged**